packages/frontend-plugin-api/src/extensions/createPageExtension.tsx

```diff
@@ -63,33 +70,49 @@ export function createPageExtension<
         ) as PortableSchema<TConfig>);
 
   return createExtension({
-    id: options.id,
-    attachTo: options.attachTo ?? { id: 'core.routes', input: 'routes' },
+    id,
+    attachTo,
+    configSchema,
+    inputs: options.inputs,
     disabled: options.disabled,
     output: {
       element: coreExtensionData.reactElement,
       path: coreExtensionData.routePath,
       routeRef: coreExtensionData.routeRef.optional(),
     },
-    inputs: options.inputs,
-    configSchema,
     factory({ bind, config, inputs, source }) {
-      const LazyComponent = React.lazy(() =>
+      const { path } = config;
+
+      const PageComponent = lazy(() =>
         options
           .loader({ config, inputs })
           .then(element => ({ default: () => element })),
       );
 
+      const ExtensionComponent = () => {
+        const analytics = useAnalytics();
+
+        // This event, never exposed to end-users of the analytics API,
+        // helps inform which extension metadata gets associated with a
+        // navigation event when the route navigated to is a gathered
+        // mountpoint.
+        useEffect(() => {
+          analytics.captureEvent(routableExtensionRenderedEvent, '');
+        }, [analytics]);
+
+        return <PageComponent />;
+      };
```

Comment:
This piece of magic makes it hard for people to implement their own `createPageExtension`-like extension factories, for example `createEntityContentExtension`. Could we bake this into the extension boundary or something like that?

Reply:
Done 🙌🏻 : https://github.com/backstage/backstage/pull/20586/commits/66af240ea692929df76d8f817e96595d648c44d6